Mathlib/LinearAlgebra/Dimension/Finrank.lean

```diff
@@ -128,3 +128,15 @@ variable (R M)
 theorem finrank_top : finrank R (⊤ : Submodule R M) = finrank R M := by
   unfold finrank
   simp
+
+section Algebra
+
+/--
+An extension of rings `R ⊆ S` is quadratic if `S` is a `R`-module of rank `2`.
+-/
+-- TODO. use this in connection with `NumberTheory.Zsqrtd`
+class IsQuadraticExtension (R S : Type*) [CommSemiring R] [Semiring S]
+  extends Algebra R S where
+  finrank_eq_two : Module.finrank R S = 2
```

Comment:
Why `Is` in the name given that this contains data?

Reply:
Oh, this is a good point!

Reply:
Indeed, I was focusing on `finrank_eq_two` but `Algebra R S` is there too. ShallI make a PR to change the name? Note though that there are also discussions (in DM) to generalize the class.

Reply:
IMHO a quick PR to at least change the name would be welcome,  in case the discussion takes forever...

Reply:
Sounds good!

Reply:
But I'm not sure what's the purpose of bundling [Algebra R S]? I think we don't usually do that.

Reply:
Yes, that might be the right solution, keep the `Is` but unbundle the `Algebra R S`

Reply:
#24369